std/net/curl.d

```diff
@@ -2261,32 +2261,46 @@ struct HTTP
     {
         p.status.reset();
 
+        CurlOption opt;
+        scope (exit)
```

Comment:
So opt is set on exit... yet the `opt` variable is changed later on. The control flow is confusing at best maybe just put p.curl.clear(opt) after the switch? Or do p.curl.xyz throw (I guess not - this is C library)?